src/mscorlib/src/System/Array.cs

```diff
@@ -976,7 +976,7 @@ public Object Clone()
             if (array==null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
             Contract.EndContractBlock();
-            return BinarySearch<T>(array, 0, array.Length, value, null);
+            return BinarySearch<T>(array, 0, array.Length, value, (IComparer<T>)null);
```

Comment:
I see that https://github.com/dotnet/coreclr/issues/2188 have `up to grab` label so made this PR.
Should I move array methods to extensions methods?  Or what should I do? I do not clear understand you wrote about because by english is not very well, sorry.